tests/pytests/integration/states/test_ansiblegate.py

```diff
@@ -59,23 +64,60 @@ def ansible_inventory(ansible_inventory_directory, sshd_server):
 
 
 @pytest.mark.requires_sshd_server
-def test_ansible_playbook(salt_call_cli, ansible_inventory):
-    ret = salt_call_cli.run(
-        "state.single",
-        "ansible.playbooks",
-        name="remove.yml",
-        git_repo="https://github.com/saltstack/salt-test-suite-ansible-playbooks.git",
-        ansible_kwargs={"inventory": ansible_inventory},
+def test_ansible_playbook(salt_call_cli, ansible_inventory, tmp_path):
+    rundir = tmp_path / "rundir"
+    rundir.mkdir(exist_ok=True, parents=True)
+    remove_contents = textwrap.dedent(
+        """
+    ---
+    - hosts: all
+      tasks:
+      - name: remove postfix
+        yum:
+          name: postfix
+          state: absent
+        become: true
+        become_user: root
+    """
     )
-    assert ret.exitcode == 0
-    assert StateResult(ret.json).result is True
-
-    ret = salt_call_cli.run(
-        "state.single",
-        "ansible.playbooks",
-        name="install.yml",
-        git_repo="https://github.com/saltstack/salt-test-suite-ansible-playbooks.git",
-        ansible_kwargs={"inventory": ansible_inventory},
+    remove_playbook = rundir / "remove.yml"
+    remove_playbook.write_text(remove_contents)
+    install_contents = textwrap.dedent(
+        """
+    ---
+    - hosts: all
+      tasks:
+      - name: install postfix
+        yum:
+          name: postfix
+          state: present
+        become: true
+        become_user: root
+    """
     )
-    assert ret.exitcode == 0
-    assert StateResult(ret.json).result is True
+    install_playbook = rundir / "install.yml"
+    install_playbook.write_text(install_contents)
+
+    # These tests have been known to timeout, so allow them longer if needed.
+    timeouts = [60, 120, 180]
+    names = ["remove.yml", "install.yml"]
+
+    for name in names:
+        for timeout in timeouts:
+            try:
+                ret = salt_call_cli.run(
+                    "state.single",
+                    "ansible.playbooks",
+                    name=name,
+                    rundir=str(rundir),
+                    ansible_kwargs={"inventory": ansible_inventory},
+                    _timeout=timeout,  # The removal can take over 60 seconds
+                )
+            except FactoryTimeout:
+                log.debug("%s took longer than %s seconds", name, timeout)
+                if timeout == timeouts[-1]:
+                    pytest.fail("Failed to run {}".format(name))
```

Comment:
**suggestion (non-blocking)** since you're breaking out of the `for` loop you can actually use the `for/else` here. `else` will run if a loop exits naturally, so:

```
for x in [1,2,3]:
    if x == 5:
        break
else:
    print('5 not found')
```

Here it will print 5 not found. Add 5 to the list and it will not print.

It's not a blocker, but it definitely could be written that way.